### PR TITLE
Homework.streams.part1.exercise2 (kovrov)

### PR DESCRIPTION
**Pull request by vkovrovEPAM**

### CODE REVIEW

File: src/test/java/streams/part1/exercise/Exercise2.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @Test
+    void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(Comparator.comparingInt(p -> p.getFullName().length()))
+                .orElse(new Person());
```
Comment (ghost) [inline src/test/java/streams/part1/exercise/Exercise2.java]: I doubt that returning a Person with default field values is a good idea. In this case you'd need to check if returned person's name is not 'default'. I would throw an exception here.

**Review by ghost — APPROVED** [review]

File: src/test/java/streams/part1/exercise/Exercise2.java (hunk 1 above)
Comment (vkovrovEPAM) [reply src/test/java/streams/part1/exercise/Exercise2.java]: Done

File: src/test/java/streams/part1/exercise/Exercise2.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+     */
+    @Test
+    void calcTotalSalaryWithCoefficientWorkExperience() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getJobHistory)
+                .map(positions -> positions.get(positions.size() - 1))
+                .map(JobHistoryEntry::getDuration)
+                .map(d -> d > 3 ? 75000d * 1.2 : 75000)
```
Comment (milkplusplus) [inline src/test/java/streams/part1/exercise/Exercise2.java:67]: Here u can use `mapToDouble(d -> d > 3 ? 75000d * 1.2 : 75000d)` and just delete the next line

File: src/test/java/streams/part1/exercise/Exercise2.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(Comparator.comparingInt(p -> p.getFullName().length()))
+                .orElseThrow(UnsupportedOperationException::new);
+
+        assertThat(expected, Matchers.is(employees.get(1).getPerson()));
+    }
+
+    @Test
+    void findEmployeeWithMaximumDurationAtOnePosition() {
```
Comment (milkplusplus) [inline src/test/java/streams/part1/exercise/Exercise2.java:43]: I wrote the same realization for this task, but then read the name carefully and realized, that it must work in another way - you should summurize "duration" for each "position" of each Employee and return the one, who has maximum "duration" (for example, you can add `new JobHistoryEntry(1, "dev", "google")` 10 times for the  `employees.get(0)` and your `Employee expected` must return him, but you current solution won't do that).

**Review by milkplusplus — APPROVED** [review]

**Pull request merged**